gtsfm/runner/gtsfm_runner_base.py

```diff
@@ -167,37 +143,27 @@ def construct_scene_optimizer(self) -> SceneOptimizer:
                 logger.info("\n\nVerifier override: " + OmegaConf.to_yaml(verifier_cfg))
                 scene_optimizer.two_view_estimator._verifier: VerifierBase = instantiate(verifier_cfg.verifier)
 
+        # Override retriever.
+        if self.parsed_args.verifier_config_name is not None:
+            with hydra.initialize_config_module(config_module="gtsfm.configs.retriever"):
+                retriever_cfg = hydra.compose(
+                    config_name=self.parsed_args.retriever_config_name,
+                )
+                logger.info("\n\nRetriever override: " + OmegaConf.to_yaml(retriever_cfg))
+                scene_optimizer.retriever: RetrieverBase = instantiate(retriever_cfg.retriever)
+        if self.parsed_args.max_frame_lookahead is not None:
+            if scene_optimizer.retriever.matching_regime in [
+                ImageMatchingRegime.SEQUENTIAL,
+                ImageMatchingRegime.SEQUENTIAL_HILTI,
+            ]:
+                scene_optimizer.retriever.max_frame_lookahead = self.parsed_args.max_frame_lookahead
+
         if self.parsed_args.mvs_off:
             scene_optimizer.run_dense_optimizer = False
 
         logger.info("\n\nSceneOptimizer: " + str(scene_optimizer))
         return scene_optimizer
 
-    def construct_retriever(self) -> RetrieverBase:
-        """Set up retriever module."""
-        matching_regime = ImageMatchingRegime(self.parsed_args.matching_regime)
-
-        if matching_regime == ImageMatchingRegime.EXHAUSTIVE:
-            retriever = ExhaustiveRetriever()
-
-        elif matching_regime == ImageMatchingRegime.RETRIEVAL:
-            retriever = NetVLADRetriever(num_matched=self.parsed_args.num_matched)
-
-        elif matching_regime == ImageMatchingRegime.SEQUENTIAL:
-            retriever = SequentialRetriever(max_frame_lookahead=self.parsed_args.max_frame_lookahead)
-
-        elif matching_regime == ImageMatchingRegime.SEQUENTIAL_HILTI:
-            retriever = SequentialHiltiRetriever(max_frame_lookahead=self.parsed_args.max_frame_lookahead)
-
-        elif matching_regime == ImageMatchingRegime.RIG_HILTI:
```

Comment:
is RIG_HILTI something we still need to support?

Reply:
yes of course!